DelayedExecution/UpdateSources/AbstractUpdateSource.cs

```diff
@@ -21,12 +27,19 @@ protected override void DisposeSelf()
         {
             UpdateHandleSourcesManager.RemoveUpdateSource(this);
             OnUpdate = null;
+
             base.DisposeSelf();
         }
 
         protected void DispatchOnUpdateEvent()
         {
+            Debug.Assert(!IsUpdating);
```

Comment:
Are these `Debug.Assert`'s necessary? 

Seems heavy handed for an update loop and might be better off as a special more intense debugging?

Reply:
It's a bit extra cautious but with so many variations on update source implementations it catches what would be a difficult to debug situation.

This performance cost is only in debug builds. If we find it's really slowing things down we can put it under a deeper debugging condition but, really, this is 2 boolean checks * a handful of update sources. So maybe an additional 10 boolean checks?